### PR TITLE
update build process

### PR DESCRIPTION
**Pull request by HuStmpHrrr**
c.f. #19

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1 +1,14 @@
-# TODO: redo this makefile properly
+EXECUTABLE:=_build/default/driver/mcltt.exe
+
+.PHONY: all clean
+
+all:
+	@cd theories; make
```
Comment (Ailrun) [inline Makefile:6]: You can use `make -C theories`

File: Makefile [hunk 2]
```diff
@@ -1 +1,14 @@
...
+	@dune build
+
+clean:
+	@cd theories; make clean
```
Comment (Ailrun) [inline Makefile:10]: Likewise

File: Makefile [hunk 3]
```diff
@@ -1 +1,14 @@
...
+	@dune clean
+	@echo "Cleaning finished."
+
+${EXECUTABLE}: all
```
Comment (Ailrun) [inline Makefile:14]: Do we need this target? It seems up-side-down in the sense that `$(EXECUTABLE)` is a part of "build all".

File: driver/Mcltt.ml [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+
+let () =
+  if Array.length Sys.argv <> 2 then
+    begin
+      Printf.fprintf stderr "incorrect input: %s <input-file>\n" Sys.argv.(0);
+      exit 1
+    end;
+  let filename = Sys.argv.(1) in
+  main ~default_fp:filename ();
+  exit 0
```
[the remark this answers is not in the document]
Comment (HuStmpHrrr) [reply driver/Mcltt.ml]: yeah no need

File: Makefile (hunk 1 above)
Comment (HuStmpHrrr) [reply Makefile]: done

File: Makefile (hunk 3 above)
Comment (HuStmpHrrr) [reply Makefile]: yeah probably not

**Review by Ailrun — APPROVED** [review]

**Pull request merged**